File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const freq = {};
	    for (let i = 0; i < nums.length; i++) {
		    if (freq[nums[i]] === undefined) {
			    freq[nums[i]] = 1;
		    } else {
			    freq[nums[i]] += 1;
		    }
	    }

	    const sorted = Object.entries(freq)
		    .sort((a, b) => b[1] - a[1])
		    .slice(0, k)
		    .map(([num]) => Number(num));

	    return sorted
    }
}

File: Data Structures & Algorithms/isomorphic-strings/submission-0.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        st = {}
        ts = {}
        i = 0
        n = len(s)-1
        while i <= n:
            if s[i] not in st:
                st[s[i]] = t[i]
            else:
                if st[s[i]] != t[i]:
                    return False

            if t[i] not in ts:
                ts[t[i]] = s[i]
            else:
                if ts[t[i]] != s[i]:
                    return False
            i+=1
        return True
            